checker/src/org/checkerframework/checker/nullness/KeyForAnnotatedTypeFactory.java

```
@@ -469,68 +469,43 @@ public AnnotationMirror createKeyForAnnotationMirrorWithValue(String value) {
    *
    */
   private LinkedHashSet<String> canonicalizeKeyForValues(AnnotationMirror anno, FlowExpressionContext flowExprContext, TreePath path, Tree t, boolean returnNullIfUnchanged) {
-      Receiver varTypeReceiver = null;
-
-      CFAbstractStore<?, ?> store = null;
-      boolean unknownReceiver = false;
-
-      if (flowExprContext.receiver == null || flowExprContext.receiver.containsUnknown()) {
-          // If the receiver is unknown, we will try local variables
-
-          store = getStoreBefore(t);
-          unknownReceiver = true; // We could use store != null for this check, but this is clearer.
-      }
-
       if (anno != null) {
+          Receiver varTypeReceiver = null;
+          boolean unknownReceiver = flowExprContext.receiver == null || flowExprContext.receiver.containsUnknown();
           boolean valuesChanged = false; // Indicates that at least one value was changed in the list.
           LinkedHashSet<String> newValues = new LinkedHashSet<String>();
 
           List<String> values = AnnotationUtils.getElementValueArray(anno, "value", String.class, false);
           for (String s: values) {
-              boolean localVariableFound = false;
-
-              if (unknownReceiver) {
-                  // If the receiver is unknown, try a local variable
-                  CFAbstractValue<?> val = store.getValueOfLocalVariableByName(s);
-
-                  if (val != null) {
-                      newValues.add(s);
-                      // Don't set valuesChanged to true since local variable names are already canonicalized
-                      localVariableFound = true;
-                  }
+              try {
+                  varTypeReceiver = FlowExpressionParseUtil.parse(s, flowExprContext, path);
+              } catch (FlowExpressionParseException e) {
```

For each empty catch block, can you add a short comment why it's OK to ignore?